### PR TITLE
Bugs(ID-216/ID-257) /App crashes when start and destination points are set to the same location and cancel option bug

### LINKED ISSUE #257: App crashes, when attempting to cancel the “Navigation Settings” sub-menu
**Describe the bug**
When clicking Cancel on the “Navigation Settings” submenu, the application crashes and exits instead of close the Navigation Settings” sub menu.

**Requirements**
- To have a mobile device
- To have Hermes app installed

**To Reproduce**
1. Open the Hermes navigation
2. Select a point in the map
3. Press on the "Navigate" blue button.
4. Press on the "Cancel"  blue button.
5. Notice that the hermes app crashes

**Expected behavior**
The “Navigation Settings” sub menu closes gracefully after pressing the Cancel button on it.

**Screenshots**
crash_on_navigation_settings

Crash
image

**Smartphone (please complete the following information):**
 - Device: Redmi Note 4
 - OS:  Android 11

### PR DESCRIPTION
**Pull request by santiCaballeroM**
## Authors: 
- Santiago Caballero Manzaneda @santiCaballeroM 

### **What?**
- Methods to handle when start and destination points are set to the same location for route calculation :heavy_check_mark: 
- An alert is shown when start/end are the same :heavy_check_mark:
- The bug "App crashes, when attempting to cancel the “Navigation Settings” sub-menu" was fixed :heavy_check_mark: 
- Navigation overlapping wapoints while choosing start point bug was fixed :heavy_check_mark: 
- Render width 4 decimals from longitude and latitude of start/end point bug was fixed to 6 decimals :heavy_check_mark: 

### **How?**

- Following MVC creating methods and xml files methods modified.

### **Why?**
- In order to have the navigation options functionality in order and the app wouldn't crash if start/end point are the same

### **Code reviewers/Testers**
@aguirrekarina 
@zrock-dev  
@saletG08 
@fabianromeroclaros 

### **Trainer (General Reviewer):**
@mmoralesv

### CODE REVIEW

**Comment by zrock-dev** [comment]
fixes #257 
closes #257

**Comment by zrock-dev** [comment]
fixes #216 
closes #216

**Review by aguirrekarina — APPROVED** [review]

**Review by zrock-dev — APPROVED** [review]

File: app/src/main/java/com/isc/hermes/controller/NavigationOptionsController.java [hunk 1]
```diff
@@ -157,11 +146,16 @@ private void manageCancelButton() {
...
             handleHiddeItemsView();
             isActive = false;
-            mapWayPointController.setMarked(false);
-            polylineRouteUpdaterController.setStartPoint(finalPoint);
+            handleCancelAction();
+            if (infoRouteController!= null) infoRouteController.deletePolylineRoutes();
         });
     }
 
+    public void handleCancelAction() {
```
Comment (saletG08) [inline app/src/main/java/com/isc/hermes/controller/NavigationOptionsController.java:157]: missing javadocs

**Review by zrock-dev — APPROVED** [review]

File: app/src/main/java/com/isc/hermes/controller/NavigationOptionsController.java (hunk 1 above)
Comment (santiCaballeroM) [reply app/src/main/java/com/isc/hermes/controller/NavigationOptionsController.java:157]: done

**Review by fabianromeroclaros — APPROVED** [review]

**Pull request merged**